src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/config/AlertRequestContext.kt

```diff
@@ -1,13 +1,27 @@
 package uk.gov.justice.digital.hmpps.hmppsalertsapi.config
 
+import org.springframework.web.context.request.RequestContextHolder
 import uk.gov.justice.digital.hmpps.hmppsalertsapi.enumeration.Source
 import uk.gov.justice.digital.hmpps.hmppsalertsapi.enumeration.Source.DPS
 import java.time.LocalDateTime
 
 data class AlertRequestContext(
-  val requestAt: LocalDateTime = LocalDateTime.now(),
-  val source: Source = DPS,
   val username: String,
   val userDisplayName: String,
-  val activeCaseLoadId: String?,
-)
+  val activeCaseLoadId: String? = null,
+  val source: Source = DPS,
+  val requestAt: LocalDateTime = LocalDateTime.now(),
+) {
+  companion object {
+    const val SYS_USER = "SYS"
+    const val SYS_DISPLAY_NAME = "Sys"
+
+    fun get(): AlertRequestContext = RequestContextHolder.getRequestAttributes()
+      ?.getAttribute(AlertRequestContext::class.simpleName!!, 0) as AlertRequestContext?
+      ?: let {
+        val context = AlertRequestContext(username = SYS_USER, userDisplayName = SYS_DISPLAY_NAME)
```

Comment:
Is this covered by AlertRequestContextConfiguration.getUserDetails in that following that call, AlertRequestContext can't be null I think.